test/unit-test/rtcp_api/rtcp_api_utest.c

```diff
@@ -0,0 +1,939 @@
+
+    receiverReport.pReceptionReports = ( RtcpReceptionReport_t * ) malloc( receiverReport.numReceptionReports * sizeof( RtcpReceptionReport_t ) );
+    for(uint32_t i = 0; i < receiverReport.numReceptionReports; i++)
+    {
+        receiverReport.pReceptionReports[i].sourceSsrc = i + 1;
+        receiverReport.pReceptionReports[i].fractionLost = 0;
+        receiverReport.pReceptionReports[i].cumulativePacketsLost = 0;
+        receiverReport.pReceptionReports[i].extendedHighestSeqNumReceived = 0;
+        receiverReport.pReceptionReports[i].interArrivalJitter = 0;
+        receiverReport.pReceptionReports[i].lastSR = 0;
+        receiverReport.pReceptionReports[i].delaySinceLastSR = 0;
+    }
+
+    result = Rtcp_SerializeReceiverReport( &context,
+                                           &receiverReport,
+                                           buffer,
+                                           &bufferLength );
+    TEST_ASSERT_EQUAL( RTCP_RESULT_OUT_OF_MEMORY,
+                       result );
+    free( receiverReport.pReceptionReports );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate RTCP DeSerialize Packet fail functionality for Bad Parameters.
+ */
+void test_rtcpDeSerializePacket_BadParams( void )
+{
+    RtcpContext_t context;
+    uint8_t serializedPacket[256];
+    size_t serializedPacketLength = sizeof( serializedPacket );
+    RtcpPacket_t rtcpPacket;
+    RtcpResult_t result;
+
+    result = Rtcp_DeserializePacket( NULL,
+                                     ( serializedPacket ),
+                                     serializedPacketLength,
+                                     &( rtcpPacket ) );
+    TEST_ASSERT_EQUAL( RTCP_RESULT_BAD_PARAM,
+                       result );
+
+    result = Rtcp_DeserializePacket( &( context ),
+                                     NULL,
+                                     serializedPacketLength,
+                                     &( rtcpPacket ) );
+    TEST_ASSERT_EQUAL( RTCP_RESULT_BAD_PARAM,
+                       result );
+
+    result = Rtcp_DeserializePacket( &( context ),
+                                     ( serializedPacket ),
+                                     serializedPacketLength,
+                                     NULL );
+    TEST_ASSERT_EQUAL( RTCP_RESULT_BAD_PARAM,
+                       result );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate RTCP DeSerialize Packet fail functionality for small Packet.
+ */
+void test_rtcpDeSerializePacket_SmallPacket( void )
+{
+    RtcpContext_t context;
+    uint8_t serializedPacket[2] = {0};
+    size_t serializedPacketLength = sizeof( serializedPacket );
+    RtcpPacket_t rtcpPacket;
+    RtcpResult_t result;
+
+    result = Rtcp_Init( &( context ) );
+    TEST_ASSERT_EQUAL( RTCP_RESULT_OK,
+                       result );
+
+
+
+    result = Rtcp_DeserializePacket( &( context ),
```

Comment:
```suggestion
    TEST_ASSERT_EQUAL( RTCP_RESULT_OK,
                       result );

    result = Rtcp_DeserializePacket( &( context ),
```
Minor suggestion - remove redundant empty lines.